### PR TITLE
update the calculate_age() function to allow for decimal age

### PR DESCRIPTION
**Pull request by Karim-Mane**

### CODE REVIEW

**Review by bahadzie — COMMENTED** [review]
The function looks ok but we should add a couple of tests to see if it works as expected.

**Review by Degoot-AM — APPROVED** [review]

File: R/span.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#'
+#' @param data The input data frame or linelist
+#' @param target_column A string used to specify the name of the date column of
+#'    interest. The values in this columns should be of type 'Date' in ISO
+#'    format ("2024-01-31").
+#' @param end_date An end date. It can be either a character that is the name of
+#'    another column of type 'Date' from the input data or a vector of Dates or
+#'    a single Date value. This should also be in the ISO format ("2024-01-31").
+#'    Default is today's date `Sys.Date()`.
+#' @param span_unit A string that specifies the units in which the spanning time
```
Comment (bahadzie) [inline R/span.R:11]: For consistency, consider changing `spanning time` to `time span` on this line and the 4 other occurences on line 14, 15 and 17

File: R/span.R [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#' Calculate time span between dates
+#'
+#' @param data The input data frame or linelist
+#' @param target_column A string used to specify the name of the date column of
+#'    interest. The values in this columns should be of type 'Date' in ISO
```
Comment (bahadzie) [inline R/span.R:5]: Can `target_column` be a vector of multiple columns? If so, consider renaming the param `target_columns` if not consider changing columns to singular.

Comment (Karim-Mane) [reply R/span.R]: fixed in commit `71ff418`

File: R/span.R (hunk 1 above)
Comment (Karim-Mane) [reply R/span.R]: fixed in commit `71ff418`

**Pull request merged**